packages/shared/src/contexts/BootProvider.tsx

```diff
@@ -179,13 +179,9 @@ export const BootDataProvider = ({
     [setBootData],
   );
 
-  const updateExperimentation = useCallback(
-    (exp: BootCacheData['exp']) => {
-      const updated = updateLocalBootData(cachedBootData, { exp });
-      setCachedBootData(updated);
-    },
-    [cachedBootData],
```

Comment:
Root cause of the issue, even though the `cachedBootData` is part of the dependencies, the callback execution was too fast after the data changed that the callback was not yet updated. Thankfully we don't need it since we can utilize using arrow function when setting state.